tests/test_dimension.py

```diff
@@ -1336,6 +1336,8 @@ def test_affiness(self):
         iterations = [i for i in FindNodes(Iteration).visit(op) if i.dim is not time]
         assert all(i.is_Affine for i in iterations)
 
+    # Skipping this test with icx, as it requires safe-math
```

Comment:
You just gave me an idea

We could rather extend `switchconfig` to swallow an additional parameter, let's call it for example `condition`. Example:

```
@switchconfig(condition=lambda i: isinstance(configuration['compiler'], OneapiCompiler),
                           safe_math=True)
```

This way we wouldn't have to skip the test

Reply:
ok

Reply:
ok